src/geometry/euclidean/basic.lean

```diff
@@ -331,6 +453,112 @@ begin
+  rw [heq, angle_eq_left] at h,
+  exact real.pi_ne_zero (by linarith),
+end
+
+/-- If ∠ABC = π then dist C B ≠ 0. -/
+lemma right_dist_ne_zero_of_angle_eq_pi {p1 p2 p3 : P} (h : ∠ p1 p2 p3 = π) : dist p3 p2 ≠ 0 :=
+begin
+  rw angle_comm at h,
+  exact left_dist_ne_zero_of_angle_eq_pi h,
+end
+
+/-- If ∠ABC = π, then (dist A C) = (dist A B) + (dist B C). -/
+lemma dist_eq_add_dist_of_angle_eq_pi {p1 p2 p3 : P} (h : ∠ p1 p2 p3 = π) :
+  (dist p1 p3) = (dist p1 p2) + (dist p3 p2) :=
+begin
+  rw [dist_eq_norm_vsub V p1 p3, dist_eq_norm_vsub V p1 p2, dist_eq_norm_vsub V p3 p2],
+  rw ← vsub_sub_vsub_cancel_right p1 p3 p2,
+  unfold angle at h,
+  exact norm_sub_eq_add_norm_of_angle_eq_pi h,
+end
+
+/-- If A ≠ B and C ≠ B then ∠ABC = π if and only if (dist A C) = (dist A B) + (dist B C). -/
+lemma dist_eq_add_dist_iff_angle_eq_pi {p1 p2 p3 : P} (hp1p2 : p1 ≠ p2) (hp3p2 : p3 ≠ p2) :
+  (dist p1 p3) = (dist p1 p2) + (dist p3 p2) ↔ ∠ p1 p2 p3 = π :=
+begin
+  rw [dist_eq_norm_vsub V p1 p3, dist_eq_norm_vsub V p1 p2, dist_eq_norm_vsub V p3 p2],
+  rw ← vsub_sub_vsub_cancel_right p1 p3 p2,
+  unfold angle,
+  exact norm_sub_eq_add_norm_iff_angle_eq_pi
+    ((λ he, hp1p2 (vsub_eq_zero_iff_eq.1 he))) (λ he, hp3p2 (vsub_eq_zero_iff_eq.1 he)),
+end
+
+/-- If ∠ABC = 0, then (dist A C) = abs ((dist A B) - (dist B C)). -/
+lemma dist_eq_abs_sub_dist_of_angle_eq_zero {p1 p2 p3 : P} (h : ∠ p1 p2 p3 = 0) :
+  (dist p1 p3) = abs ((dist p1 p2) - (dist p3 p2)) :=
+begin
+  rw [dist_eq_norm_vsub V p1 p3, dist_eq_norm_vsub V p1 p2, dist_eq_norm_vsub V p3 p2],
+  rw ← vsub_sub_vsub_cancel_right p1 p3 p2,
+  unfold angle at h,
+  exact norm_sub_eq_abs_sub_norm_of_angle_eq_zero h,
+end
+
+/-- If A ≠ B and C ≠ B then ∠ABC = 0 if and only if (dist A C) = abs ((dist A B) - (dist B C)). -/
+lemma dist_eq_abs_sub_dist_iff_angle_eq_zero {p1 p2 p3 : P} (hp1p2 : p1 ≠ p2) (hp3p2 : p3 ≠ p2) :
+  (dist p1 p3) = abs ((dist p1 p2) - (dist p3 p2)) ↔ ∠ p1 p2 p3 = 0 :=
+begin
+  rw [dist_eq_norm_vsub V p1 p3, dist_eq_norm_vsub V p1 p2, dist_eq_norm_vsub V p3 p2],
+  rw ← vsub_sub_vsub_cancel_right p1 p3 p2,
+  unfold angle,
+  exact norm_sub_eq_abs_sub_norm_iff_angle_eq_zero
+    ((λ he, hp1p2 (vsub_eq_zero_iff_eq.1 he))) (λ he, hp3p2 (vsub_eq_zero_iff_eq.1 he)),
+end
+
+/-- The midpoint of the segment AB is at the same distance from A than from B. -/
+lemma dist_left_midpoint_eq_dist_right_midpoint (p1 p2 : P) :
+  dist p1 (midpoint ℝ p1 p2) = dist p2 (midpoint ℝ p1 p2) :=
+begin
+  rw [dist_left_midpoint p1 p2, dist_right_midpoint p1 p2],
+end
+
+/-- If M is the midpoint of the segment AB, then ∠AMB = π. -/
+lemma angle_midpoint_eq_pi (p1 p2 : P) (hp1p2 : p1 ≠ p2): ∠ p1 (midpoint ℝ p1 p2) p2 = π :=
+begin
+  let m : P := midpoint ℝ p1 p2,
+  have h1 : p2 -ᵥ m = -(p1 -ᵥ m),
+  { rw [right_vsub_midpoint p1 p2, neg_vsub_eq_vsub_rev p1 m, midpoint_vsub_left p1 p2] },
+  unfold angle, rw h1,
+  refine angle_self_neg_of_nonzero _,
+  rw left_vsub_midpoint p1 p2, simp [hp1p2],
+end
```

Comment:
```suggestion
have p2 -ᵥ midpoint ℝ p1 p2 = -(p1 -ᵥ midpoint ℝ p1 p2), by { rw neg_vsub_eq_vsub_rev, simp },
by simp [angle, this, hp1p2]
```
I haven't tried it, but you can probably make similar changes in the following lemma, too